src/v/storage/compacted_index_chunk_reader.cc

```diff
@@ -131,20 +131,17 @@ compacted_index_chunk_reader::load_footer() {
       _file_size.value() - compacted_index::footer_size,
       compacted_index::footer_size,
       std::move(options));
-    ss::temporary_buffer<char> tmp = co_await in.read_exactly(
-      compacted_index::footer_size);
+    iobuf buf = co_await read_iobuf_exactly(in, compacted_index::footer_size);
 
-    if (tmp.size() != compacted_index::footer_size) {
+    if (buf.size_bytes() != compacted_index::footer_size) {
         throw std::runtime_error(fmt::format(
           "could not read enough bytes to parse "
```

Comment:
Is this the case we hit when loading old cluster's compaction index?  If so, maybe we should make the message a bit clear to indicate that this isn't totally unexpected.

Reply:
It is kind of a catch-all, but adding a note about a likely cause is a good idea.